pkg/3scale/amp/operator/system.go

```diff
@@ -223,58 +210,39 @@ func (o *OperatorSystemOptionsProvider) setResourceRequirementsOptions(b *compon
 	}
 }
 
-func (o *OperatorSystemOptionsProvider) setFileStorageOptions(b *component.SystemOptionsBuilder) {
+func (o *OperatorSystemOptionsProvider) setFileStorageOptions(b *component.SystemOptionsBuilder) error {
+	var err error
 	s3FileStorageSpec := o.APIManagerSpec.System.FileStorageSpec.S3
 	if s3FileStorageSpec != nil {
-		b.S3FileStorageOptions(component.S3FileStorageOptions{
-			AWSAccessKeyId:       "",
-			AWSSecretAccessKey:   "",
-			AWSRegion:            s3FileStorageSpec.AWSRegion,
-			AWSBucket:            s3FileStorageSpec.AWSBucket,
-			AWSCredentialsSecret: s3FileStorageSpec.AWSCredentials.Name,
-		})
+		err = o.setS3FileStorageOptions(b)
 	} else { // PVC by default
-		b.PVCFileStorageOptions(component.PVCFileStorageOptions{
-			StorageClass: o.APIManagerSpec.System.FileStorageSpec.PVC.StorageClassName,
-		})
+		o.setPVCFileStorageOptions(b)
 	}
+
+	return err
 }
 
-func (o *OperatorSystemOptionsProvider) setAWSSecretOptions(sob *component.SystemOptionsBuilder) error {
-	awsCredentialsSecretName := o.APIManagerSpec.System.FileStorageSpec.S3.AWSCredentials.Name
-	currSecret, err := getSecret(awsCredentialsSecretName, o.Namespace, o.Client)
+func (o *OperatorSystemOptionsProvider) setS3FileStorageOptions(b *component.SystemOptionsBuilder) error {
+	s3ConfigurationSecretName := o.APIManagerSpec.System.FileStorageSpec.S3.ConfigurationSecretRef.Name
+	_, err := getSecret(s3ConfigurationSecretName, o.Namespace, o.Client)
```

Comment:
We make sure the specified secret in the CR exists. We do not perform any other kind of validation at keys level nor key contents level at this moment